### PR TITLE
chore: migrate to eslint9

### PR DESCRIPTION
**Pull request by johnrazeur**
## Type

- Feature

#### What is expected?

Migrate to eslint 9.

### CODE REVIEW

File: packages/ui/.jest/a11y.test.tsx [hunk 1]
```diff
@@ -1,140 +0,0 @@
-/**
- * TODO/ outdated script with vitest migration
- **/
```
Comment (matthprost) [inline packages/ui/.jest/a11y.test.tsx:3]: Why deleting it? 😢

**Review by matthprost — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -87,7 +87,7 @@
       }
     }
   },
-  "devDependencies": {
+  "dependencies": {
```
Comment (matthprost) [inline package.json:90]: Not sure about this one too hahah

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by philibea — APPROVED** [review]

**Review by alexandre-combemorel — APPROVED** [review]

File: packages/ui/.jest/a11y.test.tsx (hunk 1 above)
Comment (philibea) [reply packages/ui/.jest/a11y.test.tsx:3]: it's not used any more, and we be replace by the next pr of axe

File: package.json [hunk 3]
```diff
@@ -107,7 +107,9 @@
     "@emotion/jest": "11.11.0",
     "@emotion/react": "11.11.4",
     "@emotion/styled": "11.11.5",
-    "@scaleway/eslint-config-react": "3.18.4",
+    "@eslint/compat": "^1.1.0",
```
Comment (matthprost) [inline package.json:110]:
```suggestion
    "@eslint/compat": "1.1.0",
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (johnrazeur) [reply package.json:90]: We should not have devDependencies in the root package.json

**Pull request merged**